src/hooks/useFormset.ts

```diff
@@ -60,10 +66,25 @@ function useFormset<TData = {}, TValue = any>(
     });
   }
 
+  const setItemMetedata = (id: string, metadata: TMetadata) => {
+    setData(data => {
+      const itemIndex = data.findIndex(item => item.id === id);
+      return [
+        ...data.slice(0, itemIndex),
+        {
+          ...data[itemIndex],
+          metadata,
+        },
+        ...data.slice(itemIndex + 1),
+      ];
```

Comment:
I'd recommend use map or reduce here, current solution is not the best in terms of speed and clearness

```suggestion
     return data.map((item) => {
        if (item. id === id) {
            return {
                ...item,
                metadata
            }
        }
         return item
     })
```

Reply:
I would recommend to split this into function - I have no clue what's happening here, the name of the function can explain a bit